### PR TITLE
Or 758 add color change functionality

### PR DESCRIPTION
**Pull request by akomiqaia**
## Describe your changes
Adds colour changing functionality which is persisted on the users' device. 

Adds Vite as a build tool and implements better dev experience. 

Implements o3-foundation and o3-button for popup styling

## Issue ticket number and link
[OR-758](https://financialtimes.atlassian.net/browse/OR-758)

[OR-758]: https://financialtimes.atlassian.net/browse/OR-758?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: tools/o3-chrome-extension/README.md [hunk 1]
```diff
@@ -42,9 +57,15 @@ For quick intro to Chrome extension components, see [Chrome Extension Tutorials]
 
 1. Clone repo
 2. `npm install`
-3. (Optional) install [Extensions Reloader](https://chrome.google.com/webstore/detail/extensions-reloader/fimgfedafeadlieiabdeeaodndnlbhid) for automatic reloading of Chrome extensions in dev mode
-4. Open Chrome and navigate to extensions (chrome://extensions/)
-5. Select 'Load unpacked extension...' and select the src folder inside o3-chrome-extension folder
+3. run `npm run dev` to start the dev server that constantly watches for changes and build files in build folder. For one time build run `npm run build`.
+4. (Optional) install [Extensions Reloader](https://chrome.google.com/webstore/detail/extensions-reloader/fimgfedafeadlieiabdeeaodndnlbhid) for automatic reloading of Chrome extensions in dev mode
+5. Open Chrome and navigate to extensions (chrome://extensions/)
+6. Select 'Load unpacked extension...' and select the dist folder from o3-chrome-extension folder
```
Comment (frshwtr) [inline tools/o3-chrome-extension/README.md:63]:
```suggestion
6. Select 'Load unpacked extension...' and select the `src` folder from o3-chrome-extension folder
```

File: tools/o3-chrome-extension/src/js/content.js [hunk 2]
```diff
@@ -1,5 +1,133 @@
-chrome.storage.local.get(['origamiGridEnabled'], function (data) {
-	if (data.origamiGridEnabled) {
-		chrome.runtime.sendMessage({enableOrigamiGrid: data.origamiGridEnabled});
+chrome.runtime.onMessage.addListener(request => {
+	if (request.message === 'creteGridOverlay') {
```
Comment (frshwtr) [inline tools/o3-chrome-extension/src/js/content.js:2]:
```suggestion
	if (request.message === 'createGridOverlay') {
```

File: tools/o3-chrome-extension/src/js/popup.js [hunk 3]
```diff
@@ -22,7 +27,128 @@ checkbox.addEventListener('change', function () {
...
+		message: 'updateGridColors',
+		colors: {
+			columnColor,
+			gutterColor,
+			marginColor,
+			colorOpacity,
+		},
+	});
+	if (partyIntervalId) {
+		// Stop the party
```
Comment (frshwtr) [inline tools/o3-chrome-extension/src/js/popup.js:109]: 🤨

**Review by frshwtr — APPROVED** [review]

File: tools/o3-chrome-extension/README.md (hunk 1 above)
Comment (akomiqaia) [reply tools/o3-chrome-extension/README.md:63]: This time it's actually a dist folder.  `npm run dev` and `npm run build` generate dist folder.

File: tools/o3-chrome-extension/src/js/content.js (hunk 2 above)
Comment (akomiqaia) [reply tools/o3-chrome-extension/src/js/content.js]: [68ca1ab](https://github.com/Financial-Times/origami/pull/1748/commits/68ca1abbd93f9898509960802cde39bc1f565e0f)

File: tools/o3-chrome-extension/src/js/popup.js (hunk 3 above)
Comment (akomiqaia) [reply tools/o3-chrome-extension/src/js/popup.js:109]: Do you think we should leave the party code here?

Comment (frshwtr) [reply tools/o3-chrome-extension/src/js/popup.js:109]: Yes, I just liked the comment 😂

**Pull request merged**